Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -35,44 +37,48 @@ This result is not yet formalised.
 * `Module.Flat.directSum`: arbitrary direct sums of flat modules are flat
 * `Module.Flat.of_free`: free modules are flat
 * `Module.Flat.of_projective`: projective modules are flat
+* `Module.Flat.iff_rTensor_preserves_injectiveness`: a module is flat iff tensoring preserves
+  injectiveness.
 
 ## TODO
 
-* Show that tensoring with a flat module preserves injective morphisms.
-  Show that this is equivalent to be flat.
-  See <https://stacks.math.columbia.edu/tag/00HD>.
-  To do this, it is probably a good idea to think about a suitable
-  categorical induction principle that should be applied to the category of `R`-modules,
-  and that will take care of the administrative side of the proof.
 * Show that flatness is stable under base change (aka extension of scalars)
   For base change, it will be very useful to have a "characteristic predicate"
   instead of relying on the construction `A ⊗ B`.
   Indeed, such a predicate should allow us to treat both
   `A[X]` and `A ⊗ R[X]` as the base change of `R[X]` to `A`.
   (Similar examples exist with `Fin n → R`, `R × R`, `ℤ[i] ⊗ ℝ`, etc...)
 * Generalize flatness to noncommutative rings.
+
 -/
 
 
 universe u v w
 
 namespace Module
 
-open Function (Injective Surjective)
+open Function (Surjective)
 
 open LinearMap Submodule TensorProduct
 
-variable (R : Type u) (M : Type v) [CommRing R] [AddCommGroup M] [Module R M]
+ variable (R : Type u) (M : Type v) [CommRing R] [AddCommGroup M] [Module R M]
 
 /-- An `R`-module `M` is flat if for all finitely generated ideals `I` of `R`,
```

Comment:
Can you add here a reference to you result? Asking because mathematicians will think about your result as the definition.

Reply:
I will add some documentation now.

Reply:
I have added some documentation